### PR TITLE
Update libcxx-setup.sh

### PR DESCRIPTION
**Pull request by madhupashish**
Here are the improvements made to the code:

1. Used `mkdir -p` to create the `llvm-build` directory. This ensures that the directory is created only if it doesn't exist.

2. Added `:` before the `${BUILD_32_BITS:=OFF}` line to prevent the script from exiting if `BUILD_32_BITS` is already set.

3. Changed `make -j cxx cxxabi unwind` to `make -j$(nproc) cxx cxxabi unwind` to utilize the number of available processors for parallel building. `$(nproc)` will provide the number of processors on most Unix-like systems.

4. Enclosed the `cmake` command arguments in an array `cmake_args` to handle arguments with spaces properly.

5. Modified the `cmake` command to use the array expansion `"${cmake_args[@]}"` to pass the arguments correctly to `cmake`. This ensures that arguments with spaces are handled correctly.

These improvements enhance the script's readability, maintainability, and performance.

### CODE REVIEW

File: .github/libcxx-setup.sh [hunk 1]
```diff
@@ -2,25 +2,27 @@
 
 set -e
 
-# Checkout LLVM sources
+Checkout LLVM sources
```
Comment (dmah42) [inline .github/libcxx-setup.sh:5]: these should still have comments i think?

**Comment by dmah42** [comment]
@madhupashish hi. are you planning on coming back to this?

**Comment by madhupashish** [comment]
> 

yes sure

**Comment by LebedevRI** [comment]
Please feel free to reopen if still relevant once addressing nits.
Thank you!

**Review by madhupashish — COMMENTED** [review]
good

**Pull request closed without merging**